File: 20-valid-parentheses/20-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    const hs = {
        '{': '}',
        '[' : ']',
        '(':')'
    }
    
    const heap = []
    
    for (let char of s){
        if(hs[char]){
            heap.push(hs[char])
        }else if( heap.pop() !== char){
           return false
        }
    }
    
    return (!heap.length)
};

File: 20-valid-parentheses/20-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        close = {"}" : "{", ")": "(", "]" : "["}
        stack = []
        
        for i in s:
            if i in close:
                if stack and stack[-1] == close[i]:
                    stack.pop()
                else:
                    return False
            else:
                stack.append(i)
                
        return True if not stack else False